project/dexter-monitor/routes/project.js

```diff
@@ -71,6 +75,68 @@ exports.saveSnapshotSummary = function() {
         })
 };
 
+exports.createProject = function(req, res) {
+
+	Promise.resolve(req.body.project)
+	.then(createDexterServerDatabase)
+	.then(insertDexterMonitorEntry)
+	.then((project) => {
+		startDexterServer(project);
+		
+		//wait for the dexter server to start
+		setTimeout( () => {
+			log.info('Dexter Server instance created at:' + project.hostName + ":" + project.portNumber );
+			res.send({status:"ok"});
+		}, 5000);
+    })
+	 .catch((err) => {
+        log.error(`Failed to create Dexter Server instance : ${err.message}`);
+        res.send({status:"fail", errorMessage: err.message});
+    });
+}
+
+function createDexterServerDatabase(project) {
+	var sql = `CREATE DATABASE ${project.projectName}; USE ${project.projectName};`;
+		sql += fs.readFileSync(`${global.config.dexterServerPath}/config/ddl.sql`).toString();	
+		sql += `USE ${global.config.database.name};`;
+	return database.exec(sql).then(() => project);
+}
+
+function insertDexterMonitorEntry(project) {
+	var sql = `INSERT INTO ProjectInfo (projectName, pid, requester, administrator, portNumber, createdDateTime, language, hostIP, projectYear, emailingWhenServerDead, groupName, dbName) VALUES ('${project.projectName}', '${project.projectName}','${project.adminName}', '${project.adminName}', '${project.portNumber}', now(), '${project.language}', '${project.hostName}', YEAR(now()), 'Y', '','${project.projectName}')`;
+	return database.exec(sql).then(() => project);
+}
+
+function startDexterServer(project) {
+	var npmInstallCmd = `npm install`;
+	var nodeCmd = `node server.js -database.name=${project.projectName} -p=${project.portNumber} -database.host=${global.config.database.host} -database.user=${global.config.database.user} -database.password=${global.config.database.password}`;
+	var dexterServerPath = path.resolve(global.config.dexterServerPath);
+	log.info("Starting new Dexter Server instance");
+	var child;
+        if (os.type()=="Linux") {
+		var supervisordEntry = `\n[program:dexter-${project.projectName}] \n` +
+					`command=` + nodeCmd + `\n` +
+					`directory=` + dexterServerPath;
+		var supervisordConfPath = '/etc/supervisord.conf';
+                try {
+		   // if supervisord is installed, dexter server will be managed by it
+		   fs.accessSync(supervisordConfPath);
+                   fs.appendFileSync(supervisordConfPath,supervisordEntry);
+		   var cmd = "supervisorctl update";
+		   child = exec(cmd); 
+		   log.info("Added Supervisor entry");
```

Comment:
Please using const, if you don't have to reassign value.

var cmd = "supervisorctl update";  child = exec(cmd); 
→ const SUPERVISORCTL_UPDATE_COMMAND=`supervisorctl update`;
child= exec(SUPERVISORCTL_UPDATE_COMMAND);